docs/new-architecture-app-intro.md

```diff
@@ -257,3 +137,371 @@ The `moduleName` has to be the same string used in the `[RCTRootView initWithBri
++                 .using(project(":ReactAndroid"))
++                 .because("On New Architecture we're building React Native from source")
++         substitute(module("com.facebook.react:hermes-engine"))
++                .using(project(":ReactAndroid:hermes-engine"))
++                .because("On New Architecture we're building Hermes from source")
++     }
++ }
+```
+
+## Android - Configure the NDK
+
+:::caution
+
+In this iteration of the guide we’re setting up the project to let you build from source. You might notice an increase in your build time because of this.
+You can mitigate this by following the approach described in [Speeding up your Build phase](/docs/next/build-speed) guide.
+
+:::
+
+As the Codegen will output some Java and some C++ code that needs to build, we need to configure the Android NDK to let you do so.
+
+Let’s edit your `android/app/build.gradle` file to include the **two** `externalNativeBuild` blocks detailed below inside the `android{}` block:
+
+```groovy
+android {
+    defaultConfig {
+        applicationId "com.awesomeproject"
+        // ...
+
+        // Add this block
+        externalNativeBuild {
+            cmake {
+                arguments "-DPROJECT_BUILD_DIR=$buildDir",
+                    "-DREACT_ANDROID_DIR=$rootDir/../node_modules/react-native/ReactAndroid",
+                    "-DREACT_ANDROID_BUILD_DIR=$rootDir/../node_modules/react-native/ReactAndroid/build",
+                    "-DANDROID_STL=c++_shared"
+            }
+        }
+    }
+
+    // Add this block
+    externalNativeBuild {
+        cmake {
+            path "$projectDir/src/main/jni/CMakeLists.txt"
+        }
+    }
+}
+```
+
+In the same `build.gradle` file, inside the same `android{}` let’s add also the following section:
+
+```groovy
+android {
+    // ...
+
+    def reactAndroidProjectDir = project(':ReactAndroid').projectDir
+    def packageReactNdkLibs = tasks.register("packageReactNdkLibs", Copy) {
+        dependsOn(":ReactAndroid:packageReactNdkLibsForBuck")
+        dependsOn("generateCodegenArtifactsFromSchema")
+        from("$reactAndroidProjectDir/src/main/jni/prebuilt/lib")
+        into("$buildDir/react-ndk/exported")
+    }
+
+    afterEvaluate {
+        preBuild.dependsOn(packageReactNdkLibs)
+        configureCMakeRelWithDebInfo.dependsOn(preReleaseBuild)
+        configureCMakeDebug.dependsOn(preDebugBuild)
+    }
+
+    packagingOptions {
+        pickFirst '**/libhermes.so'
+        pickFirst '**/libjsc.so'
+    }
+}
+```
+
+Finally, we need to create two files that are necessary for your native build to run correctly:
+
+- A CMake file with the compilation instructions (similar to a `build.gradle` for Android/Java)
+- An `OnLoad.cpp` file which, as the name says, will be loaded when your app starts.
+
+Let's create a file inside the `android/app/src/main/jni` folder called `CMakeLists.txt` with the following content:
+
+```cmake title="CMakeLists.txt"
```

Comment:
> I have the feeling that it will be easier not to miss those changes. Otherwise people will copy the whole file and may miss any required change.

There are no required changes. I've just copied over the file here.
And yes, I can suggest them to `cp` from the template, but I would rather deeplink on Github at this stage (as you can't assume that users have updated to 0.71.0).